pkg/operator/starter.go

```diff
@@ -91,15 +102,97 @@ func RunOperator(ctx context.Context, controllerConfig *controllercmd.Controller
+		operatorClient,
+		kubeInformersForNamespaces,
+		kubeClient,
+		controllerConfig.EventRecorder,
+	)
+	if err != nil {
+		return fmt.Errorf("could not create the custom CA bundle syncer: %w", err)
+	}
+
 	klog.Info("Starting the informers")
 	go kubeInformersForNamespaces.Start(ctx.Done())
 	go dynamicInformers.Start(ctx.Done())
 	go configInformers.Start(ctx.Done())
 
 	klog.Info("Starting controllerset")
 	go csiControllerSet.Run(ctx, 1)
+	go caSyncController.Run(ctx, 1)
 
 	<-ctx.Done()
 
 	return fmt.Errorf("stopped")
 }
+
+type controllerTemplateData struct {
+	CABundleConfigMap string
+}
+
+// withCustomCABundle executes the asset as a template to fill out the parts required when using a custom CA bundle.
+// The `caBundleConfigMap` parameter specifies the name of the ConfigMap containing the custom CA bundle. If the
+// argument supplied is empty, then no custom CA bundle will be used.
+func withCustomCABundle(assetFunc func(string) []byte, kubeClient kubeclient.Interface) func(string) []byte {
+	templateData := controllerTemplateData{}
+	switch used, err := isCustomCABundleUsed(kubeClient); {
+	case err != nil:
+		klog.Fatalf("could not determine if a custom CA bundle is in use: %v", err)
+	case used:
+		templateData.CABundleConfigMap = cloudConfigName
+	}
+	return func(name string) []byte {
+		asset := assetFunc(name)
+		template := template.Must(template.New("template").Parse(string(asset)))
+		buf := &bytes.Buffer{}
+		if err := template.Execute(buf, templateData); err != nil {
+			klog.Fatalf("Failed to execute ")
+		}
+		return buf.Bytes()
+	}
+}
+
+func newCustomCABundleSyncer(
+	operatorClient v1helpers.OperatorClient,
+	kubeInformers v1helpers.KubeInformersForNamespaces,
+	kubeClient kubeclient.Interface,
+	eventRecorder events.Recorder,
+) (factory.Controller, error) {
+	// sync config map with additional trust bundle to the operator namespace,
+	// so the operator can get it as a ConfigMap volume.
+	srcConfigMap := resourcesynccontroller.ResourceLocation{
+		Namespace: cloudConfigNamespace,
+		Name:      cloudConfigName,
+	}
+	dstConfigMap := resourcesynccontroller.ResourceLocation{
+		Namespace: defaultNamespace,
+		Name:      cloudConfigName,
+	}
+	certController := resourcesynccontroller.NewResourceSyncController(
+		operatorClient,
+		kubeInformers,
+		kubeClient.CoreV1(),
+		kubeClient.CoreV1(),
+		eventRecorder)
+	err := certController.SyncConfigMap(dstConfigMap, srcConfigMap)
+	if err != nil {
+		return nil, err
+	}
+	return certController, nil
+}
+
+// isCustomCABundleUsed returns true if the cloud config ConfigMap exists and contains a custom CA bundle.
+func isCustomCABundleUsed(kubeClient kubeclient.Interface) (bool, error) {
+	cloudConfigCM, err := kubeClient.CoreV1().
+		ConfigMaps(cloudConfigNamespace).
+		Get(context.Background(), cloudConfigName, metav1.GetOptions{})
+	if errors.IsNotFound(err) {
+		// no cloud config ConfigMap so there is no CA bundle
+		return false, nil
```

Comment:
I think assuming the api server will be available during the operator startup can be problematic